### DISCUSSION TITLE
Does slicing on date hierarchy ignore N/A value?

### DISCUSSION THREAD

**Discussion by HuifangYeo** [atoti/atoti#573] [Q&A]
Hi,

I have a `Result date` column as shown below: 
![image](https://user-images.githubusercontent.com/35210495/164616534-424911de-9b22-455f-8f7b-5346080766e1.png)

Slicing returns the first member of the hierarchy. However, in the above query, the first member is `N/A`.  
The slicing returns the value from 2022-04-18. Is this to be expected?

**Comment by tibdex** [comment]
I can't reproduce with atoti `0.6.5`:

```py
from datetime import date
import atoti as tt
import pandas as pd

df = pd.DataFrame(
    [
        ("abc", date(2022, 4, 25), 10),
        ("def", date(2022, 4, 26), 12),
        ("ghi", None, 15),
    ],
    columns=["ID", "Date", "Price"]
)

session = tt.Session()
table = session.read_pandas(df, table_name="Example", keys=["ID"])
cube = session.create_cube(table)
h, l, m = cube.hierarchies, cube.levels, cube.measures

h["Date"].slicing = True

get_total_price = lambda condition: cube.query(m["Price.SUM"], condition=condition)["Price.SUM"][0]

assert get_total_price(l["Date"] == "2022-04-25") == 10
assert get_total_price(l["Date"] == "N/A") == get_total_price(None) == 15
```

**Reply by HuifangYeo** [reply]
Hi,

In your use case above, the first default member has a date. If you switch the third-row data to the first row (or if you sort the date column in descending order), your default query should return 12.

Referring to https://github.com/atoti/notebooks/blob/master/notebooks/french-presidential-election/main.ipynb, it seems that the slicing will automatically assume the first valid date as the first member, ignoring the N/A.

In the jupyter notebook, we are unable to tell which date value is used for the slicing from the content editor. But from the returned query, it is returning the value corresponding to the second member.
![image](https://user-images.githubusercontent.com/35210495/165424076-a2463d0e-b92d-4ec0-9f1c-43ab651ab1c1.png)

Sorting in descending order returns the member value correctly:
![image](https://user-images.githubusercontent.com/35210495/165424696-0ed5cfcc-cdb8-4ce1-b9cc-c0ca8478c980.png)
